### ISSUE TITLE
Query: Controller actions [Save state]/[Restore state] not working?

### ISSUE THREAD

**Issue by giantclambake** [BlitterStudio/amiberry#1559]
// Discussion #1558 would demonstrate convenient timing, considering I'm currently redrafting Savestate docs  =) 

> The regular one will open the GUI, to let you select the slot.
> The Quick one, will just save to the relevant slot - there are normally 10 quick save state events, but only one is available to map to the controllers (the first one, so it will take the first slot).

...after testing in `master` ....

Controller mappings...

- Quick save state ~ saves the current state to 'Slot 0'
- Quick restore state ~ restores the saved state from 'Slot 0'

Observations: Both the above mappings seem to work as expected.

- Save state ~ does **not** open GUI as described to be able to save state
- Restore state ~ does **not** open GUI as described to be able to load/select saved state

Observations: Both the above mappings fail to action.

Note: I question if we need these last 2 mappings at all.... 

Rationale: both the `Save state` & `Restore state` mappings, can be realistically achieved by using the `Enter GUI` mapping ~ in all cases, (using the game controller), you will need to navigate to the Savestates panel, optionally select the desired 'Slot', then click on either the `Load state` or `Save state` buttons (and in the latter case, click on `Resume` to return to emulation)....

...AFAICT, the only thing the `Save state` & `Restore state` mappings could expedite, is popping the GUI with the Savestates panel already selected...(and I'm not sure if that's possible anyway 8)....

*If one does use the `Enter GUI` mapping instead, what may help expedite matters, would be a `Resume emulation` mapping... TBH I'm surprised we don't already have that, as a synonym for 'Exit GUI'...all other GUI buttons are mapped, why not this too?

So for mine ... remove the custom mappings `Save state` & `Restore state` --> add a `Resume emulation` mapping ---> as I'm aware of all this, it can be covered off by details in docs...

Thoughts?

TIA

**Comment by midwan** [comment]
Those shortcuts are not yet implemented in `amiberry_gui.cpp`, so they don't currently do anything, even though the events are triggered normally.
It's something to add for the future, I think.

**Comment by giantclambake** [comment]
Yeah, right you are.... there's an effective workaround, so it's no biggy ~ docs are easily updated when it comes time ;)

**Comment by giantclambake** [comment]
....just adding these so they're not forgotten...

Shift + END + F6  will prompt a "Save save state" dialog
END + F6 will prompt a "Restore save state" one

Neither of the above key-sequences work (with the GUI)

> I don't see a keyboard shortcut for quick loading a savestate, from what I can see

This would be handy, to compliment quick saving keyboard shortcut (which works as expected)

**Comment by midwan** [comment]
Bear in mind, that with 624df66cd8cdf81cd1b6276bfac00bf5f94e6396 we now have per-config savestate path as well.

**Comment by midwan** [comment]
This is now implemented, with a few new shortcuts added:
- End + F5 = opens a dialog to select a CD image to insert in the drive
- End + F6 = opens a dialog to select a save state to load
- End + Shift + F6 = opens a dialog to select a save state to save

Keep in mind that saving a statefile with this method does not automatically assign it to a slot, unless you follow the same naming convention manually, of course. You can give it any filename you want.
